main/mafia.hs

```diff
@@ -502,6 +509,18 @@ mafiaScript :: File -> [Argument] -> EitherT MafiaError IO ()
 mafiaScript file args =
   firstT MafiaScriptError $ runScript file args
 
+mafiaExec :: [Argument] -> EitherT MafiaError IO ()
+mafiaExec args = do
+  let fixedArgs =
+        case args of
+          [] -> [] -- Should not happen.
+          [x] -> [x] -- A command without arguments.
+          -- Insert `--` between the command and its arguments so cabal doesn't
+          -- mess with them.
+          (x:xs) -> x : "--" : xs
+  liftCabal $ cabal_ "exec" fixedArgs
```

Comment:
Sorry, I missed this the first time, you probably want to use `Mafia.Process.exec` here rather than `cabal_`, so that we exec the `cabal` executable rather than just calling out to it.